generators/client/templates/angular/src/main/webapp/app/shared/language/_language.helper.ts

```diff
@@ -23,25 +24,30 @@ export class JhiLanguageHelper {
      * 3. 'global.title'
      */
     updateTitle(titleKey?: string) {
-
-        if (!titleKey && this.titleService.getTitle() ) {
-            titleKey = this.titleService.getTitle();
+        if (!titleKey) {
+             titleKey = this.getPageTitle(this.router.routerState.snapshot.root);
         }
 
-        this.translateService.get(titleKey || 'global.title').subscribe(title => {
+        this.translateService.get(titleKey).subscribe(title => {
             this.titleService.setTitle(title);
         });
     }
 
     private init () {
-        <%_
-        // TODO fix this part
-        // this.translateService.onTranslationChange.subscribe((event: TranslationChangeEvent) => {
-        //     this.updateTitle();
-        // });
-        // this.translateService.onLangChange.subscribe((event: LangChangeEvent) => {
-        //     this.updateTitle();
-        // });
-        _%>
+        this.translateService.onTranslationChange.subscribe((event: TranslationChangeEvent) => {
+            this.updateTitle();
+        });
+
+        this.translateService.onLangChange.subscribe((event: LangChangeEvent) => {
+            this.updateTitle();
+        });
+    }
+
+    private getPageTitle(routeSnapshot: ActivatedRouteSnapshot) {
+        let title: string = (routeSnapshot.data && routeSnapshot.data['pageTitle']) ? routeSnapshot.data['pageTitle'] : 'ang4App';
```

Comment:
I fixed it via [this](https://github.com/jhipster/generator-jhipster/commit/11495631ff45c1f51f5083f2b6e8b9f970972467)